src/main/java/org/embulk/base/restclient/RestClientInputPluginBaseUnsafe.java

```diff
@@ -91,24 +89,27 @@ public TaskReport run(TaskSource taskSource, Schema schema, int taskIndex, PageO
         T task = taskSource.loadTask(this.taskClass);
         ServiceResponseMapper serviceResponseMapper =
             this.serviceResponseMapperBuilder.buildServiceResponseMapper(task);
+
+        TaskReport taskReport = null;
         try (PageBuilder pageBuilder = new PageBuilder(Exec.getBufferAllocator(), schema, output)) {
             try (AutoCloseableClient<T> clientWrapper = new AutoCloseableClient<T>(task, this.clientCreator)) {
                 RetryHelper retryHelper = new RetryHelper(
                     clientWrapper.getClient(),
                     task.getRetryLimit(),
                     task.getInitialRetryWait(),
                     task.getMaxRetryWait());
-                this.serviceDataIngester.ingestServiceData(task,
-                                         retryHelper,
-                                         serviceResponseMapper.createRecordImporter(),
-                                         taskIndex,
-                                         pageBuilder);
+                taskReport = this.serviceDataIngester.ingestServiceData(
```

Comment:
Couldn't we put `return` statement here? We also could call `checkNotNull` method here to avoid NPE.